lib/outdated.js

```diff
@@ -1,437 +1,243 @@
-    }
-    if (!long) {
-      return cb(null, [[tree, dep, c, wanted, latest, req, null, pkgpath]])
-    }
-    cb(null, [[tree, dep, c, wanted, latest, req, type, pkgpath]])
+  if (opts.color) {
+    columns[0] = color[current === wanted ? 'yellow' : 'red'](columns[0]) // current
+    columns[2] = color.green(columns[2]) // wanted
+    columns[3] = color.magenta(columns[3]) // latest
   }
 
-  function updateLocalDeps (latestRegistryVersion) {
-    fetchPackageMetadata('file:' + parsed.fetchSpec, '.', (er, localDependency) => {
-      if (er) return cb()
-
-      var wanted = localDependency.version
-      var latest = localDependency.version
-
-      if (latestRegistryVersion) {
-        latest = latestRegistryVersion
-        if (semver.lt(wanted, latestRegistryVersion)) {
-          wanted = latestRegistryVersion
-          req = dep + '@' + latest
-        }
-      }
+  return columns
+}
 
-      if (!curr || curr.version !== wanted) {
-        doIt(wanted, latest)
-      } else {
-        skip()
-      }
-    })
-  }
+// --parseable creates output like this:
+// <fullpath>:<name@wanted>:<name@installed>:<name@latest>
+function makeParseable (list, opts) {
+  return list.map(dep => {
+    const { name, current, wanted, latest, path, type, homepage } = dep
+    const out = [
+      path,
+      name + '@' + wanted,
+      current ? (name + '@' + current) : 'MISSING',
+      name + '@' + latest
+    ]
+    if (opts.long) out.push(type, homepage)
 
-  function updateDeps (er, d) {
-    if (er) return cb(er)
+    return out.join(':')
+  }).join(os.EOL)
+}
 
-    if (parsed.type === 'alias') {
-      req = parsed.subSpec.rawSpec
-    }
-    try {
-      var l = pickManifest(d, 'latest')
-      var m = pickManifest(d, req)
-    } catch (er) {
-      if (er.code === 'ETARGET' || er.code === 'E403') {
-        return skip(er)
-      } else {
-        return skip()
-      }
+function makeJSON (list, opts) {
+  const out = {}
+  list.forEach(dep => {
+    const { name, current, wanted, latest, path, type, homepage } = dep
+    out[name] = {
+      current,
+      wanted,
+      latest,
+      location: path
     }
-
-    // check that the url origin hasn't changed (#1727) and that
-    // there is no newer version available
-    var dFromUrl = m._from && url.parse(m._from).protocol
-    var cFromUrl = curr && curr.from && url.parse(curr.from).protocol
-
-    if (
-      !curr ||
-      (dFromUrl && cFromUrl && m._from !== curr.from) ||
-      m.version !== curr.version ||
-      m.version !== l.version
-    ) {
-      if (parsed.type === 'alias') {
-        doIt(
-          `npm:${parsed.subSpec.name}@${m.version}`,
-          `npm:${parsed.subSpec.name}@${l.version}`
-        )
-      } else {
-        doIt(m.version, l.version)
-      }
-    } else {
-      skip()
+    if (opts.long) {
+      out[name].type = type
+      out[name].homepage = homepage
     }
-  }
+  })
+  return JSON.stringify(out, null, 2)
 }
```

Comment:
we might want to stick with that new cmd export pattern from [`lib/install.js`](https://github.com/npm/cli/blob/release/v7.0.0-beta/lib/install.js), [`lib/ci.js`](https://github.com/npm/cli/blob/release/v7.0.0-beta/lib/ci.js), etc - e.g:

```
const usage = usageUtil('outdated', 'npm outdated')
const completion = (cb) => cb(null, [])
const cmd = (args, cb) => outdated().then(() => cb()).catch(cb)

const outdated = async () => {
  ...impl
}

module.exports = Object.assign(cmd, { completion, usage })
```